### PR TITLE
add limit to cachedMatcher

### PR DESCRIPTION
**Pull request by ilyam8**

### CODE REVIEW

File: pkg/matcher/cache.go [hunk 1]
```diff
@@ -1,31 +1,83 @@
...
+		} else {
+			cm.cache = newLRUCache(limit)
 		}
+		return cm
 	}
 }
 
 func (m *cachedMatcher) Match(b []byte) bool {
 	s := string(b)
+	if m.cache == nil {
```
Comment (Wing924) [inline pkg/matcher/cache.go]: it never be true because `m.cache` must not be `nil`

File: pkg/matcher/cache.go [hunk 2]
```diff
@@ -35,6 +87,10 @@ func (m *cachedMatcher) Match(b []byte) bool {
 }
 
 func (m *cachedMatcher) MatchString(s string) bool {
+	if m.cache == nil {
```
Comment (Wing924) [inline pkg/matcher/cache.go]: it never be true because m.cache must not be nil

File: pkg/matcher/cache.go [hunk 3]
```diff
@@ -1,31 +1,86 @@
...
-import "sync"
+import (
+	"sync"
+
+	"github.com/hashicorp/golang-lru"
+)
+
+type cache interface {
+	Get(key string) (result bool, exist bool)
+	Len() int
```
Comment (Wing924) [inline pkg/matcher/cache.go]: do we need `Len()`?

File: modules/activemq/activemq.go [hunk 4]
```diff
@@ -29,10 +29,11 @@ var (
 )
 
 var (
-	defMaxQueues   = 50
-	defMaxTopics   = 50
-	defURL         = "http://127.0.0.1:8161"
-	defHTTPTimeout = time.Second
+	defMaxQueues       = 50
```
Comment (Wing924) [inline modules/activemq/activemq.go]: `def` usually means not `default` but `define`.
I think use `default` is more readable.

**Comment by ilyam8** [comment]
thx @Wing924

**Pull request merged**